corehq/warehouse/transforms/sql/form_fact.sql

```diff
@@ -0,0 +1,44 @@
+INSERT INTO {{ form_fact }} (
+    form_id,
+    domain,
+    domain_dim_id,
+    app_id,
+    xmlns,
+    user_id,
+    user_dim_id,
+    received_on,
+    deleted_on,
+    edited_on,
+    build_id,
+    state,
+    last_modified
+)
+SELECT
+    form_id,
+    domain_table.domain,
+    domain_table.id,
+    app_id,
+    xmlns,
+    user_table.user_id,
+    user_table.id,
+    received_on,
+    deleted_on,
+    edited_on,
+    build_id,
+    state,
+    CASE
+        WHEN
+            (received_on > deleted_on OR deleted_on IS NULL) AND
+            (received_on > edited_on OR edited_on IS NULL)
+        THEN received_on::timestamp AT TIME ZONE default_timezone
+        WHEN
+            (deleted_on > received_on OR received_on IS NULL) AND
+            (deleted_on > edited_on OR edited_on IS NULL)
+        THEN deleted_on::timestamp AT TIME ZONE default_timezone
+        ELSE edited_on::timestamp AT TIME ZONE default_timezone
```

Comment:
ah yes `GREATEST ` exactly what i was looking for.

yes this stores the form in the domain's local timezone. as for if this will be the only one we'll be querying on, that's a good question. i wonder if we need to even worry about `deleted_on` and `edited_on`. Mostly likely we'll want to aggregate always by `received_on`. even for reports like "submissions by day" i don't think we actually care if the form was archived or edited